### ISSUE TITLE
Apply failing with API request timed out when creating multiple resources of the same type with a for_each

### ISSUE THREAD

**Issue by janritter** [gessnerfl/terraform-provider-instana#191]
## Problem
When trying to create multiple application configs (application perspectives) in with a for_each loop and the instana_application_config resource. The first resource creation always succeeds immediately, with the second one failing after the 30 seconds timeout. Another apply immediately creates the second resource successfully. Looking at the debug logs it looks like there is only an API call for the first resource creation and none for the second one. 
The same happened with instana_alerting_channel on apply and destroy.

I will also try to have a look at the code and see if I can find the problem and create a PR.

## Terraform code
```hcl
resource "instana_application_config" "this" {
  for_each = { for perspective in var.application_perspectives : perspective.label => perspective }

  label          = each.value.label
  scope          = each.value.scope
  boundary_scope = each.value.boundary_scope
  tag_filter     = each.value.tag_filter
}

# Var-file
application_perspectives = [
  {
    label          = "my-app-2"
    scope          = "INCLUDE_ALL_DOWNSTREAM"
    boundary_scope = "INBOUND"
    tag_filter     = "agent.tag:stage EQUALS 'dev'"
  },
  {
    label          = "your-app-2"
    scope          = "INCLUDE_NO_DOWNSTREAM"
    boundary_scope = "DEFAULT"
    tag_filter     = "aws.ec2.tag:stage EQUALS 'test' OR call.tag:stage@na EQUALS 'test'"
  }
]
```

## Plan output
```text
Terraform will perform the following actions:

  # instana_application_config.this["my-app-2"] will be created
  + resource "instana_application_config" "this" {
      + boundary_scope = "INBOUND"
      + id             = (known after apply)
      + label          = "my-app-2"
      + scope          = "INCLUDE_ALL_DOWNSTREAM"
      + tag_filter     = "agent.tag:stage@dest EQUALS 'dev'"
    }

  # instana_application_config.this["your-app-2"] will be created
  + resource "instana_application_config" "this" {
      + boundary_scope = "DEFAULT"
      + id             = (known after apply)
      + label          = "your-app-2"
      + scope          = "INCLUDE_NO_DOWNSTREAM"
      + tag_filter     = "(aws.ec2.tag:stage@dest EQUALS 'test' OR call.tag:stage@na EQUALS 'test')"
    }

Plan: 2 to add, 0 to change, 0 to destroy.
```

## Apply debug output
```text
2023-09-22T14:06:10.065+0200 [INFO]  provider: configuring client automatic mTLS
2023-09-22T14:06:10.072+0200 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/gessnerfl/instana/2.0.0/darwin_amd64/terraform-provider-instana_v2.0.0 args=[.terraform/providers/registry.terraform.io/gessnerfl/instana/2.0.0/darwin_amd64/terraform-provider-instana_v2.0.0]
2023-09-22T14:06:10.076+0200 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/gessnerfl/instana/2.0.0/darwin_amd64/terraform-provider-instana_v2.0.0 pid=23530
2023-09-22T14:06:10.076+0200 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/gessnerfl/instana/2.0.0/darwin_amd64/terraform-provider-instana_v2.0.0
2023-09-22T14:06:10.088+0200 [INFO]  provider.terraform-provider-instana_v2.0.0: configuring server automatic mTLS: timestamp=2023-09-22T14:06:10.088+0200
2023-09-22T14:06:10.105+0200 [DEBUG] provider.terraform-provider-instana_v2.0.0: plugin address: address=/var/folders/w7/9vh2ry750jb562py4v5ckqth0000gq/T/plugin1832553054 network=unix timestamp=2023-09-22T14:06:10.105+0200
2023-09-22T14:06:10.105+0200 [DEBUG] provider: using plugin: version=5
instana_application_config.this["my-app-2"]: Creating...
instana_application_config.this["your-app-2"]: Creating...
2023-09-22T14:06:10.330+0200 [INFO]  provider.terraform-provider-instana_v2.0.0: 2023/09/22 14:06:10 [DEBUG] Call PUT https://<redacted>.instana.io/api/application-monitoring/settings/application/ck6o6cn70jnlnqk2o3fg: timestamp=2023-09-22T14:06:10.330+0200
instana_application_config.this["my-app-2"]: Creation complete after 1s [id=ck6o6cn70jnlnqk2o3fg]
instana_application_config.this["your-app-2"]: Still creating... [10s elapsed]
instana_application_config.this["your-app-2"]: Still creating... [20s elapsed]
instana_application_config.this["your-app-2"]: Still creating... [30s elapsed]
2023-09-22T14:06:40.135+0200 [ERROR] provider.terraform-provider-instana_v2.0.0: Response contains error diagnostic: tf_proto_version=5.4 tf_provider_addr=provider tf_req_id=52628d86-69be-fed5-bc5a-e024f9e45b6c tf_rpc=ApplyResourceChange diagnostic_detail= diagnostic_severity=ERROR diagnostic_summary="API request timed out" tf_resource_type=instana_application_config @caller=github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/diag/diagnostics.go:58 @module=sdk.proto timestamp=2023-09-22T14:06:40.135+0200
╷
│ Error: API request timed out
│ 
│   with instana_application_config.this["your-app-2"],
│   on main.tf line 15, in resource "instana_application_config" "this":
│   15: resource "instana_application_config" "this" {
│ 
╵
2023-09-22T14:06:40.998+0200 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-09-22T14:06:41.000+0200 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/gessnerfl/instana/2.0.0/darwin_amd64/terraform-provider-instana_v2.0.0 pid=23530
2023-09-22T14:06:41.000+0200 [DEBUG] provider: plugin exited
terraform apply returned the following error code: exit status 1
```

**Comment by gessnerfl** [comment]
@janritter this issue is related to a recent change related to throttling in the API client.

**Comment by gessnerfl** [comment]
Fixed with version 2.0.2
